### PR TITLE
Improve README files

### PR DESCRIPTION
**Pull request by erikd**
Add a new README for `cardano-sl-core`.

Replace existing README for `cardano-sl-lib`. The old README was an exact duplicate of the file one directory level up.

### CODE REVIEW

**Comment by erikd** [comment]
Happy to get corrections or suggestions for improvements on any of these READMEs.

File: binary/README.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+Binary serialisation and deserialisation for Cardano SL.
+
+This library is built on top of the [binary] and [cborg] packages and implement
+space efficient serialising and deserialising of Haskell data types to and from
+binary representations via a `Bi` type class and associated machinery.
+
+The library includes ability the derive `Bi` instances (requires a `Typeable`
+constraint).
+
+[binary]: https://hackage.haskell.org/package/binary
```
Comment (dcoutts) [inline binary/README.md]: I can see it's there. Are those instances still used?

Comment (erikd) [reply binary/README.md]: Well the `deriveSimpleBi`*  functions are used in at least 6 other packages. Maybe the derived instances are not used, but that's another matter.

Comment (erikd) [reply binary/README.md]: Are you saying that the `Bi` typeclass shouldn't actually be used?

I started at the top of the tree in `explorer` and tried to remove usage of `deriveSimpleBi` but it seems pretty baked in and still very much in use.

I also noticed that the binary package seems to define two ways of generate `Bi` instances, the TH  method and one using generics. I had a bit of a stab at moving from TH to generics in `explorer` but its non-trivial so I'm dropping that for now.

**Comment by gromakovsky** [comment]
Now main README in repository is this: https://i.imgur.com/XZmI9wm.png

**Comment by erikd** [comment]
Top level README restored in #2655.

**Pull request merged**